packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -21,24 +21,62 @@ export default function ({ types: t }) {
     return false;
   }
 
-  function createObjectSpread(file, props, objRef) {
-    const restElement = props.pop();
-
+  // extracts ["a", "b"] from {a: "hi", b}
+  function extractKeys(path) {
+    const propPaths = path.get("properties");
     const keys = [];
-    for (const prop of props) {
-      let key = prop.key;
-      if (t.isIdentifier(key) && !prop.computed) {
-        key = t.stringLiteral(prop.key.name);
+    for (const propPath of propPaths) {
+      let key = propPath.get("key").node;
+      if (t.isIdentifier(key) && !propPath.node.computed) {
```

Comment:
Numerical literals get returned as-is and are handled later using the `toPropertyKey` helper.

The purpose of this function is to just convert identifier keys to string literals e.g. `["a", "b"] from {a: "hi", b}`

renaming it to `extractNormalizedKeys` and adding more comments to explain the purpose of the function